### ISSUE TITLE
M2A querying in rest filter returns error

### ISSUE THREAD

**Issue by rijkvanzanten** [directus/directus#23179]
Querying using the `:` syntax in the rest API throws a permissions error:

```
{"errors":[{"message":"You don't have permission to access field \"item:basic_collection\" in collection \"example_m2a\" or it does not exist. Queried in \"m2a\".","extensions":{"code":"FORBIDDEN","reason":"You don't have permission to access field \"item:basic_collection\" in collection \"example_m2a\" or it does not exist. Queried in \"m2a\".","stack":"DirectusError: You don't have permission to access field \"item:basic_collection\" in collection \"example_m2a\" or it does not exist. Queried in \"m2a\".\n    at createFieldsForbiddenError (/directus/api/src/permissions/modules/process-ast/utils/validate-path/create-error.ts:16:9)\n    at validatePathExistence (/directus/api/src/permissions/modules/process-ast/utils/validate-path/validate-path-existence.ts:16:9)\n    at processAst (/directus/api/src/permissions/modules/process-ast/process-ast.ts:28:4)\n    at ItemsService.readByQuery (/directus/api/src/services/items.ts:500:15)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at ItemsService.readOne (/directus/api/src/services/items.ts:564:19)\n    at <anonymous> (/directus/api/src/controllers/items.ts:109:18)"}}]}
```

**Comment by rijkvanzanten** [comment]
Note

```
http://localhost:8055/items/example/91bfd1fb-e0c9-4ea7-8380-8a03e6e10b2b?fields=m2a.item:basic_collection.id
```

does work as expected (with the extra nested path at the end)

**Comment by hanneskuettner** [comment]
Fixed by #23181
